src/ui/base-page.js

```diff
@@ -0,0 +1,155 @@
+/*
+Copyright 2018-2019 OCAD University
+Licensed under the New BSD license. You may not use this file except in compliance with this licence.
+You may obtain a copy of the BSD License at
+https://raw.githubusercontent.com/fluid-project/sjrk-story-telling/master/LICENSE.txt
+*/
+
+/* global fluid, sjrk */
+
+"use strict";
+
+(function ($, fluid) {
+
+    fluid.defaults("sjrk.storyTelling.base.page", {
+        gradeNames: ["fluid.modelComponent"],
+        model: {
+            uiLanguage: "en" //initial state is English
+        },
+        pageSetup: {
+            resourcePrefix: ""
+            // "authoringEnabled" is retrieved from sjrk.storyTelling.server.config.json5
+            // via a request to "/clientConfig". It enables and disables the
+            // authoring capabilities of the tool and must be present.
+            //authoringEnabled: true
+        },
+        distributeOptions: {
+            "ui.templateManager.authoringEnabled": {
+                source: "{that}.options.pageSetup.authoringEnabled",
+                target: "{that ui templateManager}.options.model.dynamicValues.authoringEnabled"
+            },
+            "ui.templateManager.resourcePrefix": {
+                source: "{that}.options.pageSetup.resourcePrefix",
+                target: "{that ui}.options.components.templateManager.options.templateConfig.resourcePrefix"
```

Comment:
This and the next target selector are overlong and fragile - in general it's a mistake to distribute to a tail path that is not attached to the same component. Absorb these into the component selector.